### PR TITLE
ProjectReference, Reference, and package creation properties

### LINKED ISSUE #18195: Document ProjectReference and referencing .dlls
`ProjectReference` to reference a project.

To reference an assembly:

```
<ItemGroup>
  <Reference Include="MyAssembly">
    <HintPath>path\to\MyAssembly.dll</HintPath>
  </Reference>
</ItemGroup>
```

https://medium.com/@tonerdo/referencing-a-net-dll-directly-using-the-net-core-toolchain-16f0af46a4dc

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 59d4dafc-2217-9a7b-4590-27faa85164cd
* Version Independent ID: 64b80e80-1c15-c293-466a-918de98e96f6
* Content: [MSBuild properties for Microsoft.NET.Sdk - .NET Core](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props)
* Content Source: [docs/core/project-sdk/msbuild-props.md](https://github.com/dotnet/docs/blob/master/docs/core/project-sdk/msbuild-props.md)
* Product: **dotnet-core**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by gewarren**
- Fixes #18195
- Removes extraneous lines from project file snippets
- Fleshes out the package creation and restoration sections

### CODE REVIEW

File: docs/core/project-sdk/msbuild-props.md [hunk 1]
```diff
@@ -165,147 +163,165 @@ You can configure some run-time behaviors by specifying MSBuild properties in th
...
-- [PackageReference](#packagereference)
 - [AssetTargetFallback](#assettargetfallback)
+- [PackageReference](#packagereference)
+- [ProjectReference](#projectreference)
+- [Reference](#reference)
+- [Restore properties](#restore-properties)
+
+### AssetTargetFallback
+
+The `AssetTargetFallback` property lets you specify additional compatible framework versions for projects that your project references and NuGet packages that your project consumes. For example, if you specify a package dependency using `PackageReference` but that package doesn't contain assets that are compatible with your projects's `TargetFramework`, the `AssetTargetFallback` property comes into play. The compatibility of the referenced package is rechecked using each target framework that's specified in `AssetTargetFallback`.
```
Comment (tdykstra) [inline docs/core/project-sdk/msbuild-props.md:261]: Suggestion to simplify the first sentence

```suggestion
The `AssetTargetFallback` property lets you specify additional compatible framework versions for project references and NuGet packages. For example, if you specify a package dependency using `PackageReference` but that package doesn't contain assets that are compatible with your projects's `TargetFramework`, the `AssetTargetFallback` property comes into play. The compatibility of the referenced package is rechecked using each target framework that's specified in `AssetTargetFallback`.
```

File: docs/core/project-sdk/msbuild-props.md [hunk 2]
```diff
@@ -165,147 +163,165 @@ You can configure some run-time behaviors by specifying MSBuild properties in th
...
+</ItemGroup>
 ```
 
 For more information, see [Package references in project files](/nuget/consume-packages/package-references-in-project-files).
 
-### AssetTargetFallback
+### ProjectReference
 
-The `AssetTargetFallback` property lets you specify additional compatible framework versions for projects that your project references and NuGet packages that your project consumes. For example, if you specify a package dependency using `PackageReference` but that package doesn't contain assets that are compatible with your projects's `TargetFramework`, the `AssetTargetFallback` property comes into play. The compatibility of the referenced package is rechecked using each target framework that's specified in `AssetTargetFallback`.
+The `ProjectReference` defines a reference to another project. The referenced project is added as a NuGet package dependency, that is, it's treated the same as a `PackageReference`.
```
Comment (tdykstra) [inline docs/core/project-sdk/msbuild-props.md:289]:
```suggestion
The `ProjectReference` element defines a reference to another project. The referenced project is added as a NuGet package dependency, that is, it's treated the same as a `PackageReference`.
```

File: docs/core/project-sdk/msbuild-props.md [hunk 3]
```diff
@@ -165,147 +163,165 @@ You can configure some run-time behaviors by specifying MSBuild properties in th
...
+</ItemGroup>
+```
+
+### Restore properties
+
+Restoring a referenced package installs all of its direct dependencies and all the dependencies of those dependencies. You can customize package restoration by specifying properties such as `RestorePackagesPath` and `RestoreIgnoreFailedSources`. For more information about these and other properties, see [restore target](/nuget/reference/msbuild-targets#restore-target).
+
+```xml
+<PropertyGroup>
+  ...
```
Comment (tdykstra) [inline docs/core/project-sdk/msbuild-props.md:321]: Just curious why you have ellipses in some PropertyGroup examples like this one but not in others.

**Review by tdykstra — APPROVED** [review]
LGTM, a couple suggestions and a question.

File: docs/core/project-sdk/msbuild-props.md (hunk 3 above)
Comment (gewarren) [reply docs/core/project-sdk/msbuild-props.md:321]: I removed this one but left the one on line 67 because that is where <TargetFramework> needs to go.

**Comment by gewarren** [comment]
Thanks for the review, Tom!

**Pull request merged**